core/startup/config_file_watcher.go

```diff
@@ -73,11 +73,13 @@ func WatchConfigDirectory(configDir string, appConfig *config.ApplicationConfig)
 				if !ok {
 					return
 				}
-				if event.Has(fsnotify.Write) {
+				if event.Has(fsnotify.Write) || event.Has(fsnotify.Create) || event.Has(fsnotify.Rename) {
 					for targetName, watchFn := range CONFIG_FILE_UPDATES {

```

Comment:
rather than looping over the entire map every time an fsnotify event happens, we should attempt to directly pull the correct handler from the map:
```
					watchFn, ok := CONFIG_FILE_UPDATES[path.Base(event.Name)]
					if !ok {
						log.Debug().Str("filename", event.Name).Msg("no configuration file handler found for file")
						continue
					}
```
After that block runs, we will know that we have the handler inside watchFn, and we don't have to range over the entire map to do it.  Not a big deal while there are only 2 files, but if this ends up scaling to be larger (potentially managing the model configs to allow them to be updated at runtime) that map could get considerably larger.